addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjection.java

```diff
@@ -0,0 +1,687 @@
+        if (doAllDataScan) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(
+                        "\nStarting with the url-encoded query injection package of attack rules:");
+            }
+            int index = 0;
+            for (String valueInj : ALL_DATA_VALUE_INJECTION) {
+                String paramInj = param + ALL_DATA_PARAM_INJECTION[index++];
+                if (isStop()) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(STOP_LOG);
+                    }
+                    return;
+                }
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("\nTrying with the value: " + paramInj + valueInj);
+                }
+                try {
+                    msgInjAttack = getNewMsg();
+                    setParameter(msgInjAttack, paramInj, valueInj);
+                    sendAndReceive(msgInjAttack, false);
+                    rtt.add(msgInjAttack.getTimeElapsedMillis());
+                    bodyInjAttack = msgInjAttack.getResponseBody().toString();
+                    if (msgInjAttack.getResponseHeader().getStatusCode() != HttpStatusCode.OK) {
+                        continue;
+                    }
+                    if (!bodyBase.equals(bodyInjAttack)) {
+                        counterProofMsg = getNewMsg();
+                        setParameter(counterProofMsg, param + ZAP_QUERYSTRING, valueInj);
+                        sendAndReceive(counterProofMsg, false);
+                        bodyCounterProof = counterProofMsg.getResponseBody().toString();
+                        if (bodyBase.equals(bodyCounterProof)) {
+                            bingo(
+                                    Alert.RISK_HIGH,
+                                    Alert.CONFIDENCE_HIGH,
+                                    getName(),
+                                    getDescription(),
+                                    null,
+                                    param,
+                                    paramInj + valueInj,
+                                    getExtraInfo(ALL_DATA_ATTACK),
+                                    getSolution(),
+                                    msgInjAttack);
+                            isBingo = true;
+                            break;
+                        }
+                    }
+                } catch (IOException ex) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(
+                                "Caught "
+                                        + ex.getClass().getName()
+                                        + " "
+                                        + ex.getMessage()
+                                        + " when "
+                                        + URI_EX_LOG);
+                    }
+                    return;
+                }
+            }
+        }
+        // search for not-handled errors
+        if (doCrashScan) {
```

Comment:
Worth checking `isBingo` as well (same for following tests), no need to test further if we already know there's an injection.